admin-dev/themes/new-theme/js/pages/carrier/form/components/CarrierRangesModal.vue

```diff
@@ -0,0 +1,374 @@
+            this.overlappingAlert = true;
+          }
+
+          // Check min < max for each range
+          if (range.max !== null && range.min !== null && range.max <= range.min) {
+            table.querySelectorAll(`tr[data-row="${index}"] input.form-max`)
+              .forEach((input) => {
+                input.classList.add('is-invalid');
+              });
+            this.errors = true;
+          }
+
+          saveMax = range.max;
+        });
+      },
+      addRange(index: undefined|number) {
+        // Add new range at the index specified, at the bottom if not specified
+        // (with "min" already set to the previous "max")
+        if (index === undefined) {
+          this.ranges.push({min: this.ranges[this.ranges.length - 1]?.max, max: null});
+        } else {
+          this.ranges.splice(index + 1, 0, {min: this.ranges[index]?.max, max: null});
+        }
+      },
+      deleteRange(rangeIndex: number) {
+        // We remove the selected range
+        this.ranges.splice(rangeIndex, 1);
+        // We add an empty range if there is none
+        if (this.ranges.length === 0) {
+          this.ranges.push({min: null, max: null});
+        }
+      },
+      dragStart(rangeIndex: number) {
+        // We save the current range index to know which one is being dragged
+        this.dragIndex = rangeIndex;
+      },
+      dragOver(event: DragEvent, rangeIndex: number) {
+        // We retreive the row being dragged and the row being hovered
+        const table = <HTMLElement> document.querySelector('.table-carrier-ranges-modal');
+        const row = <HTMLElement> table.querySelector(`tr[data-row="${this.dragIndex}"]`);
+        const rowHover = <HTMLElement> table.querySelector(`tr[data-row="${rangeIndex}"]`);
+
+        // We move the dragged row before or after the hovered row depending on the position of the mouse on it
+        if (row !== null && rowHover !== null) {
+          if (event.offsetY > rowHover.offsetHeight / 2) {
+            rowHover.parentNode?.insertBefore(row, rowHover.nextSibling);
+          } else {
+            rowHover.parentNode?.insertBefore(row, rowHover);
+          }
+        }
+      },
+      dragEnd() {
+        // We reorder the ranges according to the new order when the drag ending
+        const rangesNewOrder:Range[] = [];
+        const rows = document.querySelectorAll('.table-carrier-ranges-modal tbody tr[data-row]');
+        rows.forEach((row) => {
+          const rowId = row.getAttribute('data-row');
+
+          if (rowId !== null) {
+            rangesNewOrder.push(this.ranges[parseInt(rowId, 10)]);
+          }
+        });
+
+        // We update the ranges and +1 to the refreshKey to force the refresh
+        this.ranges.splice(0, this.ranges.length, ...rangesNewOrder);
+        this.dragIndex = null;
+        this.refreshKey += 1;
+      },
+    },
+  });
+</script>
+
+<style lang="scss" type="text/scss">
+  @import '~@scss/config/_settings.scss';
+
+  .modal {
```

Comment:
Have we done the same in other components? I am worried it could make it hard to customize the style

Reply:
I don't know the context, but styling .modal globally is not a good idea, but maybe it's inside some other selector?

Reply:
To me, is ok to do this because is specific to this component. But, as @Hlavtox said, we must no doing this globally. In fact, I need to `scope` this style for this modal only. Vuejs permit us to scope the style by add `scoped` in `<style>` tag. (In background, vuejs automatically add unique attribute for scoping style for this component)